onnxscript/onnx_types.py

```diff
@@ -36,131 +33,144 @@ def check_shape(shape):
         check_dim(shape)
 
 
-class TensorType:
-    """ONNX Script representation of a tensor type."""
+tensor_type_registry: dict[DType, "TensorType"] = {}
+tensor_type_shape_cache: dict[DType, "TensorType"] = {}
```

Comment:
Good catch. I had moved these out of `TensorType` originally since I didn't want them accessible on the subclasses. In that context quoting is necessary.